modules/benchmark/R/check_if_legal_table.R

```diff
@@ -0,0 +1,66 @@
+#' @title check_if_legal_table
+#' @details
+#' Checks if custom_table:
+#' 1. is formated correctly
+#' 2. is complete (has all of the species/pft's in both observations)
+#' 3. is condense-able (Could be represented as a hierachry)
+#' 
+#' @param observation_one a vector of plant fucntional types, or species
+#' @param observation_two anouther vector of plant fucntional types, or species
+#' @param custom_table a table that either maps two pft's to one anouther or maps custom species codes to bety id codes. 
+#' In the second case, must be passable to match_species_id. 
+#' @return \code{boolean} 
+#' @author Tempest McCabe
+check_if_legal_table<-function(table, observation_one, observation_two){
+  all_there<-TRUE
+  names<-names(table)
+  if(!"plant_functional_type_one" %in% names|!"plant_functional_type_two" %in% names ){
+    logger.severe("Custom table provided does not use correct column names. Requires both 'plant_fucntional_type_one', and 'plant_fucntional_type_two'. 
+                  Column names are currently", names(table))
+  }else{
+    missing<-list()
+    for(h in 1:length(observation_one)){
+      if(!observation_one[h] %in% table$plant_functional_type_one){all_there<-FALSE; missing<-c(missing,observation_one[h])}
```

Comment:
Again, extend this out into a block. As a general rule, you should almost always avoid putting multiple statements on the same line (i.e. use semicolons like this).